crypto/cmp/cmp_msg.c

```diff
@@ -984,8 +984,7 @@ static int suitable_rid(const ASN1_INTEGER *certReqId, int rid)
         return 1;
 
     trid = ossl_cmp_asn1_get_int(certReqId);
-
-    if (trid == OSSL_CMP_CERTREQID_NONE) {
+    if (trid < OSSL_CMP_CERTREQID_NONE) {
```

Comment:
== NONE was not ok before - but it is now?

Reply:
Good question. It took me a while to remember:

I made this change to reflect that the error return code of the internal function `ossl_cmp_asn1_get_int()`
had changed in commit 2c8d9f19e351a84d4329fbe2f68a4a8a49cad3ef from -1 to -2,
thus the comparison for the error situation had to be adapted which I had overlooked back then.
So I changed the error detection to `OSSL_CMP_CERTREQID_INVALID` (which has value -2)
and also to cover anything further less than `OSSL_CMP_CERTREQID_NONE` (which as value -1).

Yet to make this more clear, now changed the comparison to the equivalent `trid <= OSSL_CMP_CERTREQID_INVALID`.